Server/mods/deathmatch/logic/packets/CEntityAddPacket.cpp

```diff
@@ -959,6 +959,25 @@ bool CEntityAddPacket::Write(NetBitStreamInterface& BitStream) const
                         BitStream.Write(currentWeaponSlot);
                     }
 
+                    // Aniamtion
+                    if (BitStream.Can(eBitStreamVersion::AnimationsSync))
+                    {
+                        const SPlayerAnimData& animData = pPed->GetAnimationData();
+
+                        if (!animData.blockName.empty() && !animData.animName.empty())
+                        {
+                            BitStream.WriteString(animData.blockName);
```

Comment:
You have to write a **bit** that signalling whether this packet contains animation data or not.

Reply:
What is the difference in this case?

Reply:
Because you have a packet corruption now.

Reply:
I understand, the idea is that later data will be incorrect due to "holes" when reading?

Reply:
Yes, without that **bit** a client doesn't know what exactly is written inside a packet. Because a string itself doesn't contain sufficient information. You should explicitly mark it.